lib/msal-node/src/network/HttpClient.ts

```diff
@@ -75,6 +75,12 @@ const networkRequestViaProxy = <T>(
         tunnelRequestOptions.timeout = timeout;
     }
 
+    if (options?.customAgentOptions && Object.keys(options.customAgentOptions).length) {
+        tunnelRequestOptions.agent = new http.Agent({
```

Comment:
Generally param reassignment like this is a no-no. You should create a new local variable if you need to modify properties

Reply:
I'm modifying a new local variable that I created.